packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/tile_overlay.dart

```diff
@@ -0,0 +1,159 @@
+/// west to east and the y values range from 0 to 2N - 1 and increase from north to south.
+///
+class TileOverlay {
+  /// Creates an immutable representation of a [TileOverlay] to draw on [GoogleMap].
+  const TileOverlay({
+    @required this.tileOverlayId,
+    this.fadeIn = true,
+    this.tileProvider,
+    this.transparency = 0.0,
+    this.zIndex,
+    this.visible = true,
+    this.tileSize = 256,
+  }) : assert(transparency >= 0.0 && transparency <= 1.0);
+
+  /// Uniquely identifies a [TileOverlay].
+  final TileOverlayId tileOverlayId;
+
+  /// Whether the tiles should fade in. The default is true.
+  final bool fadeIn;
+
+  /// The tile provider to use for this tile overlay.
+  final TileProvider tileProvider;
+
+  /// The transparency of the tile overlay. The default transparency is 0 (opaque).
+  final double transparency;
+
+  /// The tile overlay's zIndex, i.e., the order in which it will be drawn where
+  /// overlays with larger values are drawn above those with lower values
+  final int zIndex;
+
+  /// The visibility for the tile overlay. The default visibility is true.
+  final bool visible;
+
+  /// Specifies the number of pixels (not points) that the returned tile images will prefer
+  /// to display as. iOS only.
+  ///
+  /// Defaults to 256, which is the traditional size of Google Maps tiles.
+  /// As an example, an application developer may wish to provide retina tiles (512 pixel edge length)
+  /// on retina devices, to keep the same number of tiles per view as the default value of 256
+  /// would give on a non-retina device.
+  final int tileSize;
+
+  /// Creates a new [Polygon] object whose values are the same as this instance,
+  /// unless overwritten by the specified parameters.
+  TileOverlay copyWith({
+    TileOverlayId tileOverlayId,
+    bool fadeInParam,
+    double transparencyParam,
+    int zIndexParam,
+    bool visibleParam,
+    int tileSizeParam,
+  }) {
+    return TileOverlay(
+      tileOverlayId: tileOverlayId,
+      fadeIn: fadeInParam ?? fadeIn,
+      transparency: transparencyParam ?? transparency,
+      zIndex: zIndexParam ?? zIndex,
+      visible: visibleParam ?? visible,
+      tileSize: tileSizeParam ?? tileSize,
+    );
+  }
+
+  /// Converts this object to JSON.
+  dynamic toJson() {
+    final Map<String, dynamic> json = <String, dynamic>{};
+
+    void addIfPresent(String fieldName, dynamic value) {
+      if (value != null) {
+        json[fieldName] = value;
+      }
+    }
+
+    addIfPresent('tileOverlayId', tileOverlayId.value);
+    addIfPresent('fadeIn', fadeIn);
+    addIfPresent('transparency', transparency);
+    addIfPresent('zIndex', zIndex);
+    addIfPresent('visible', visible);
+    addIfPresent('tileSize', tileSize);
+
+    return json;
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+    if (other.runtimeType != runtimeType) return false;
+    final TileOverlay typedOther = other;
```

Comment:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#common-boilerplates-for-operator--and-hashcode

Reply:
done